src/app/onion/learning-object-builder/components/info-page/metadata/metadata.component.ts

```diff
@@ -30,4 +65,52 @@ export class LearningObjectMetadataComponent implements OnInit {
       this.learningObject.addLevel(level);
     }
   }
+
+  search() {
+    this.query.text = this.query.text.trim();
+      this.userService.searchUsers(this.query.text).then(val => {
+        this.users = JSON.parse(val);
+        this.tempArrayOfKeys = Object.keys(this.users);
+         // Limit search results to 10
+         if (this.tempArrayOfKeys.length <= 10) {
+          this.arrayOfKeys = this.tempArrayOfKeys;
+         } else {
+            for (let i = 0; i < 10; i++) {
+              this.arrayOfKeys[i] = this.tempArrayOfKeys[i];
+            }
+          }
+          // If query is empty, remove previous results
+          if (this.query.text === '') {
+            this.arrayOfKeys = [];
+            this.tempArrayOfKeys = [];
+          }
+      });
+  }
+
+  addAuthor(index) {
+    if (this.isAuthorSelected) {
+      // this.selectedAuthors.push(this.users[index]._username);
+      this.selectedAuthors.push(this.users[index]._username);
```

Comment:
Should be able to use `.username` getter method if object is instantiated as a User entity. Best to keep consistency in the way we're accessing properties on entities.